src/python/turicreate/data_structures/serialization.py

```diff
@@ -0,0 +1,51 @@
+
+
+# SFrames require a disk-backed file or directory to work with.  This directory 
+# has to be present to allow for serialization or deserialization. 
+__serialization_directory = None 
+
+
+def enable_sframe_serialization(serialization_directory):
+    """
+    Enables pickling of sframes through the use of a user-set directory to 
+    store the objects.  This directory must be present and set through 
+    this method for deserialization to work, but may be a different directory.
+    """
+
+    import os
+
+    global __serialization_directory
+    if serialization_directory is None:
+        __serialization_directory = None
+        return 
+
+    __serialization_directory = os.path.abspath(os.path.expanduser(serialization_directory))
+
+    # Make sure the directory exists.
+    if not os.path.exists(__serialization_directory):
+        raise ValueError("Directory %s does not exist." % __serialization_directory)
+
+    # Is it a directory? 
+    if not os.path.isdir(__serialization_directory):
+        raise ValueError("%s is not a directory." % __serialization_directory)
+
+
+
+def serialization_directory():
+    """
+    Returns the current serialization directory if set, or None otherwise. 
+    """
+    global __serialization_directory
```

Comment:
You don't need this line in your function unless you're modifying the value of the global variable.

Reply:
It's not needed but it is good practice, so I always do it.  It avoids confusing issues such as reading the variable, then assigning to it, and then those referring to two different variables.

Reply:
Interesting. I've never heard that before, but it makes sense.